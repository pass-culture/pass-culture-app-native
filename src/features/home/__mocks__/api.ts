import { beneficiaryUser } from 'fixtures/user'

export const useUserProfileInfo = jest.fn().mockReturnValue({
  data: beneficiaryUser,
})
