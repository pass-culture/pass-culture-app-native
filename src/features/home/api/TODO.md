# TODO

## Links

- [PC-33065](https://passculture.atlassian.net/browse/PC-33065)
- [MobTime](https://mobtime.hadrienmp.fr/mob/pass-culture)

---

## Tasks

- [ ] Mocker useActivationBanner dans HomeBanner et NonBeneficiaryBanner + clean les tests 

- [ ] Écrire des tests pour useActivationBanner

- [ ] Ajouter une nouvelle condition dans useActivationBanner pour savoir si l'utilisateur dois voir ou non le titre "Débloque tes ...€"

---

## Tasks for another US

- [ ] 