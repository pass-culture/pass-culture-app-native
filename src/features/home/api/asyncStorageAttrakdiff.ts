import AsyncStorage from '@react-native-async-storage/async-storage'

import { Attrakdiff } from './useAttrakdiffModal'

export const createAsyncStorageAttrakdiff = (key: string): Attrakdiff => {
  return {
    hasTriggered: async () => (await AsyncStorage.getItem(key)) === 'true',
    setTriggered: async () => {
      await AsyncStorage.setItem(key, 'true')
    },
  }
}
