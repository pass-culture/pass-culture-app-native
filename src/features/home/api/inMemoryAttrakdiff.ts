import { Attrakdiff } from './useAttrakdiffModal'

export const createInMemoryAttrakdiff = (): Attrakdiff => {
  let hasTriggered = false
  return {
    hasTriggered: async () => hasTriggered,
    setTriggered: async () => {
      hasTriggered = true
    },
  }
}
