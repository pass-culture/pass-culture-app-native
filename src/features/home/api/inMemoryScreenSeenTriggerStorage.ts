import { ScreenSeenCount, ScreenSeenCountTriggerStorage } from '../helpers/getScreenSeenCount'

export const createInMemoryScreenSeenCountTriggerStorage = (): ScreenSeenCountTriggerStorage => {
  const triggered: number[] = []

  return {
    hasTriggered: async (screenSeenCount: ScreenSeenCount) => triggered.includes(screenSeenCount),
    setTriggered: async (screenSeenCount: ScreenSeenCount) => {
      triggered.push(screenSeenCount)
    },
  }
}
