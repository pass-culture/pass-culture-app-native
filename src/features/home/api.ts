import resolveResponse from 'contentful-resolve-response'
import { useMemo, useEffect } from 'react'
import { useQuery } from 'react-query'

import { api } from 'api/api'
import { UserProfileResponse } from 'api/gen'
import { useAuthContext } from 'features/auth/AuthContext'
import { NoContentError } from 'features/home/components/NoContentError'
import {
  HomepageEntry,
  EntryCollection,
  EntryFields,
  processHomepageEntry,
} from 'features/home/contentful'
import { useSelectPlaylist } from 'features/home/selectPlaylist'
import { analytics } from 'libs/analytics'
import { env } from 'libs/environment'
import { getExternal } from 'libs/fetch'
import { ScreenError } from 'libs/monitoring'
import { useNetwork } from 'libs/network/useNetwork'
import { QueryKeys } from 'libs/queryKeys'

const DEPTH_LEVEL = 2

export const CONTENTFUL_BASE_URL = 'https://cdn.contentful.com'
export const BASE_URL = `${CONTENTFUL_BASE_URL}/spaces/${env.CONTENTFUL_SPACE_ID}/environments/${env.CONTENTFUL_ENVIRONMENT}`
export const PARAMS = `?include=${DEPTH_LEVEL}&content_type=homepageNatif&access_token=${env.CONTENTFUL_ACCESS_TOKEN}`

export async function getEntries() {
  const url = `${BASE_URL}/entries${PARAMS}`
  try {
    const json = await getExternal<EntryCollection<EntryFields, 'homepageNatif'>>(url)
    return resolveResponse(json)
  } catch (e) {
    const error = e as Error
    throw new ScreenError(error?.message, NoContentError)
  }
}

export function useHomepageModules(paramsEntryId?: string) {
  const { isConnected } = useNetwork()
  const selectPlaylist = useSelectPlaylist(paramsEntryId)
  const { data: entries } = useQuery<HomepageEntry[]>(QueryKeys.HOMEPAGE_MODULES, getEntries, {
    enabled: isConnected,
  })

  const entry = selectPlaylist(entries || [])
  const entryId = entry?.sys.id

  useEffect(() => {
    if (entryId) analytics.logConsultHome({ entryId })
  }, [entryId])

  return useMemo(() => (entry ? processHomepageEntry(entry) : []), [entryId])
}

export function useUserProfileInfo(options = {}) {
  const { isConnected } = useNetwork()
  const { isLoggedIn } = useAuthContext()

  return useQuery<UserProfileResponse>(QueryKeys.USER_PROFILE, () => api.getnativev1me(), {
    enabled: isLoggedIn && isConnected,
    ...options,
  })
}
