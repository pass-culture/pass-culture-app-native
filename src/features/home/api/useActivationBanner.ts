import { CurrencyEnum } from 'api/gen'
import { useAuthContext } from 'features/auth/context/AuthContext'
import { useBanner } from 'features/home/api/useBanner'
import { useGetStepperInfo } from 'features/identityCheck/api/useGetStepperInfo'
import { useFeatureFlag } from 'libs/firebase/firestore/featureFlags/useFeatureFlag'
import { RemoteStoreFeatureFlags as featureFlags } from 'libs/firebase/firestore/types'
import { GeolocPermissionState, useLocation } from 'libs/location'
import { LocationMode } from 'libs/location/types'

export function useActivationBanner() {
  const enablePacificFrancCurrency = useFeatureFlag(featureFlags.ENABLE_PACIFIC_FRANC_CURRENCY)
  const { user } = useAuthContext()
  const { selectedLocationMode, permissionState } = useLocation()
  const { data: subscription } = useGetStepperInfo()

  const nextSubscriptionStepEnable = !!subscription?.nextSubscriptionStep
  const isEligibleForBeneficiaryUpgrade = !!user?.isEligibleForBeneficiaryUpgrade
  const isActivationProcessEnable = nextSubscriptionStepEnable || isEligibleForBeneficiaryUpgrade

  const isUserLocated = selectedLocationMode !== LocationMode.EVERYWHERE
  const isUserRegisteredInPacificFrancRegion = user?.currency === CurrencyEnum.XPF
  const isUserInRelevantLocation = isUserLocated || isUserRegisteredInPacificFrancRegion

  const isGeolocated = permissionState === GeolocPermissionState.GRANTED
  const { data } = useBanner(isGeolocated)

  if (enablePacificFrancCurrency) {
    if (isActivationProcessEnable || isUserInRelevantLocation) {
      return {
        banner: {
          title: data?.banner?.title,
          text: data?.banner?.text,
          name: data?.banner?.name,
        },
      }
    }
  }
  return { banner: { ...data?.banner } }
}
