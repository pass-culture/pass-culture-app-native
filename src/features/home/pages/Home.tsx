import { t } from '@lingui/macro'
import { useFocusEffect, useNavigation, useRoute } from '@react-navigation/native'
import React, { FunctionComponent, useState } from 'react'
import { NativeSyntheticEvent, NativeScrollEvent, ScrollView, TouchableOpacity } from 'react-native'
import { getStatusBarHeight } from 'react-native-iphone-x-helper'
import styled from 'styled-components/native'

import { useUserProfileInfo } from 'features/home/api'
import { HomeBodyPlaceholder, SignUpSignInChoiceModal } from 'features/home/components'
import { HomeBody } from 'features/home/components/HomeBody'
import { useDisplayedHomeModules } from 'features/home/pages/useDisplayedHomeModules'
import { useAvailableCredit } from 'features/home/services/useAvailableCredit'
import { UseNavigationType, UseRouteType } from 'features/navigation/RootNavigator'
import { analytics } from 'libs/analytics'
import { isCloseToBottom } from 'libs/analytics.utils'
import { env } from 'libs/environment'
import { _ } from 'libs/i18n'
import { formatToFrenchDecimal } from 'libs/parsers'
import { useModal } from 'ui/components/modals/useModal'
import { HeaderBackground } from 'ui/svg/HeaderBackground'
import { UserCircle } from 'ui/svg/icons/UserCircle'
import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'
import { ACTIVE_OPACITY } from 'ui/theme/colors'
import { ZIndexes } from 'ui/theme/layers'

import { useShowSkeleton } from './useShowSkeleton'

const statusBarHeight = getStatusBarHeight(true)

export const Home: FunctionComponent = function () {
  const navigation = useNavigation<UseNavigationType>()
  const { params } = useRoute<UseRouteType<'Home'>>()
  const { data: userInfos } = useUserProfileInfo()
  const { visible: signInModalVisible, showModal: showSignInModal, hideModal } = useModal(false)
  const showSkeleton = useShowSkeleton()
  const availableCredit = useAvailableCredit()

  const [hasSeenAllModules, setHasSeenAllModules] = useState<boolean>(false)
  const { displayedModules, algoliaModules } = useDisplayedHomeModules()

  function hideSignInModal() {
    navigation.setParams({ shouldDisplayLoginModal: false })
    hideModal()
  }

  useFocusEffect(() => {
    if (params.shouldDisplayLoginModal) {
      showSignInModal()
    }
  })

  const checkIfAllModulesHaveBeenSeen = ({
    nativeEvent,
  }: NativeSyntheticEvent<NativeScrollEvent>) => {
    if (!hasSeenAllModules && isCloseToBottom(nativeEvent)) {
      setHasSeenAllModules(true)
      analytics.logAllModulesSeen(displayedModules.length)
    }
  }

  let subtitle = _(t`Toute la culture dans ta main`)
  if (availableCredit) {
    subtitle = availableCredit.isExpired
      ? _(t`Ton crédit est expiré`)
      : _(t`Tu as ${formatToFrenchDecimal(availableCredit.amount)} sur ton pass`)
  }

  return (
    <ScrollView
      testID="homeScrollView"
      scrollEventThrottle={400}
      bounces={false}
      onScroll={checkIfAllModulesHaveBeenSeen}>
      <Spacer.TopScreen />
      {env.CHEAT_BUTTONS_ENABLED && (
        <CheatButtonsContainer>
          <CheatTouchableOpacity onPress={() => navigation.navigate('AppComponents')}>
            <Typo.Body>{_(t`Composants`)}</Typo.Body>
          </CheatTouchableOpacity>
          <CheatTouchableOpacity onPress={() => navigation.navigate('Navigation')}>
            <Typo.Body>{_(t`Navigation`)}</Typo.Body>
          </CheatTouchableOpacity>
        </CheatButtonsContainer>
      )}
      <HeaderBackgroundWrapper>
        <HeaderBackground />
      </HeaderBackgroundWrapper>
      <UserProfileContainer onPress={() => navigation.navigate('CheatMenu')}>
        <UserCircle size={32} color={ColorsEnum.WHITE} />
      </UserProfileContainer>

      <CenterContainer>
        <Spacer.Column numberOfSpaces={8} />
        <StyledTitle1 color={ColorsEnum.WHITE} numberOfLines={2}>
          {userInfos?.firstName
            ? _(/*i18n: Hello title message */ t`Bonjour ${userInfos?.firstName}`)
            : _(/*i18n: Welcome title message */ t`Bienvenue !`)}
        </StyledTitle1>
        <Spacer.Column numberOfSpaces={2} />
        <Typo.Body color={ColorsEnum.WHITE}>{subtitle}</Typo.Body>
      </CenterContainer>
      <Spacer.Column numberOfSpaces={6} />

      {showSkeleton ? <HomeBodyPlaceholder /> : null}
      <HomeBodyLoadingContainer isLoading={showSkeleton}>
        <HomeBody modules={displayedModules} algoliaModules={algoliaModules} />
      </HomeBodyLoadingContainer>

      <SignUpSignInChoiceModal visible={signInModalVisible} dismissModal={hideSignInModal} />
      <Spacer.TabBar />
    </ScrollView>
  )
}

const StyledTitle1 = styled(Typo.Title1)({
  textAlign: 'center',
  marginHorizontal: getSpacing(8),
})

const CenterContainer = styled.View({
  flex: 1,
  alignItems: 'center',
})

const HeaderBackgroundWrapper = styled.View({
  position: 'absolute',
  top: 0,
  left: 0,
  zIndex: 0,
})

const HomeBodyLoadingContainer = styled.View<{ isLoading: boolean }>(({ isLoading }) => ({
  height: isLoading ? 0 : undefined,
  overflow: 'hidden',
}))

const CheatButtonsContainer = styled.View({
  width: '100%',
  flexDirection: 'row',
  justifyContent: 'space-around',
  zIndex: ZIndexes.HomeCheatButton,
  top: 20,
})

const UserProfileContainer = styled.TouchableOpacity.attrs({
  activeOpacity: ACTIVE_OPACITY,
})({
  position: 'absolute',
  right: 24,
  top: getSpacing(3) + statusBarHeight,
  zIndex: 1,
})

const CheatTouchableOpacity = styled(TouchableOpacity)({
  borderColor: ColorsEnum.BLACK,
  borderWidth: 2,
  padding: getSpacing(1),
})
