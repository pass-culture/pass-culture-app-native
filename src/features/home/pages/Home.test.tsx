import { render, fireEvent, waitFor } from '@testing-library/react-native'
import React from 'react'

import * as authApi from 'features/auth/api'
import { env } from 'libs/environment'

import { Home } from './Home'

jest.mock('libs/environment', () => ({
  env: {
    FEATURE_FLAG_CODE_PUSH: true,
  },
}))

function mockUseCurrentUser({
  email = 'john.doe@passculture.app',
  isFetching = false,
  isError = false,
  refetch = async () => 'john.doe@passculture.app',
}: Partial<ReturnType<typeof authApi.useCurrentUser>> = {}): void {
  const value = { email, isFetching, isError, refetch, error: undefined }
  jest.spyOn(authApi, 'useCurrentUser').mockReturnValue(value)
}

describe('Home component', () => {
  const navigation = {
    navigate: jest.fn(),
  } as any // eslint-disable-line @typescript-eslint/no-explicit-any

  afterAll(() => jest.resetAllMocks())

  describe('With current user loaded', () => {
    beforeEach(() => {
      mockUseCurrentUser()
    })

    it('should have a welcome message', async () => {
      const { getByText } = render(<Home navigation={navigation} />)

      const welcomeText = await waitFor(() => getByText('Bienvenue à Pass Culture'))
      expect(welcomeText.props.children).toBe('Bienvenue à Pass Culture')
    })

    it('should render correctly', async () => {
      const home = render(<Home navigation={navigation} />)
      expect(home).toMatchSnapshot()
    })

    it('should have a button to go to login page with params', () => {
      const { getByText } = render(<Home navigation={navigation} />)
      const LoginButton = getByText('Aller sur la page de connexion avec params')
      fireEvent.press(LoginButton)
      expect(navigation.navigate).toHaveBeenCalledWith('Login', {
        userId: 'I have been Set by params',
      })
    })

    it('should not have code push button', async () => {
      env.FEATURE_FLAG_CODE_PUSH_MANUAL = false
      const home = render(<Home navigation={navigation} />)
      expect(() => home.getByText('Check update')).toThrowError()
    })
  })

  describe('Without current user', () => {
    it('should display activity indicator when loading', () => {
      mockUseCurrentUser({ isFetching: true })

      const { getByTestId } = render(<Home navigation={navigation} />)
      const activityIndicator = getByTestId('user-activity-indicator')
      expect(activityIndicator).toBeTruthy()
    })

    it('should not display activity indicator when not loading', () => {
      mockUseCurrentUser({ isFetching: false })

      const { getByTestId } = render(<Home navigation={navigation} />)
      let activityIndicator
      try {
        activityIndicator = getByTestId('user-activity-indicator')
        fail()
      } catch (error) {
        expect(activityIndicator).toBeFalsy()
      }
    })

    it('should not display user view on error', () => {
      mockUseCurrentUser({ isError: true })

      const { getByText } = render(<Home navigation={navigation} />)
      let refreshButton
      try {
        refreshButton = getByText("Rafraîchir les données de l'utilisateur")
        fail()
      } catch (error) {
        expect(refreshButton).toBeFalsy()
      }
    })

    it('should call refetch when the refetchbutton is clicked', () => {
      const refetch = jest.fn()
      mockUseCurrentUser({ refetch })

      const { getByTestId } = render(<Home navigation={navigation} />)
      const refreshButton = getByTestId('refreshUserInformationsButton')
      refreshButton.props.onClick()

      expect(refetch).toBeCalled()
    })
  })
})
