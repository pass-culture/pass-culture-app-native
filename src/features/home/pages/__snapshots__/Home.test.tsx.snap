// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component With current user loaded should render correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="none"
          bbHeight={352}
          bbWidth={754}
          focusable={false}
          height={352}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 352,
                "width": 754,
              },
            ]
          }
          vbHeight={352}
          vbWidth={375}
          width={754}
        >
          <RNSVGGroup>
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  Array [
                    0,
                    -1376171,
                    1,
                    -13500266,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="prefix__b"
                x1="34.782%"
                x2="88.023%"
                y1="5.945%"
                y2="111.119%"
              />
              <RNSVGLinearGradient
                gradient={
                  Array [
                    0,
                    -1376171,
                    1,
                    -13500266,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="prefix__c"
                x1="50%"
                x2="37.519%"
                y1="15.944%"
                y2="116.187%"
              />
              <RNSVGPath
                d="M0 0h375v352H0z"
                name="prefix__a"
              />
            </RNSVGDefs>
            <RNSVGGroup
              fill={null}
              fillRule={0}
              propList={
                Array [
                  "fill",
                  "fillRule",
                ]
              }
            >
              <RNSVGMask
                fill={4294967295}
                height="100%"
                maskContentUnits={1}
                maskTransform={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                maskUnits={0}
                name="prefix__d"
                propList={
                  Array [
                    "fill",
                  ]
                }
                width="100%"
                x="0%"
                y="0%"
              >
                <RNSVGUse
                  height={0}
                  href="prefix__a"
                  width={0}
                  x={0}
                  y={0}
                />
              </RNSVGMask>
              <RNSVGUse
                fill={
                  Array [
                    1,
                    "prefix__b",
                  ]
                }
                height={0}
                href="prefix__a"
                propList={
                  Array [
                    "fill",
                  ]
                }
                width={0}
                x={0}
                y={0}
              />
              <RNSVGPath
                d="M89.727-4.98c24.421 187.472 211.753 121.594 197.5 277.418-4.24 46.354-22.177 69.072-46.982 79.563L0 351.989V-5z"
                fill={
                  Array [
                    1,
                    "prefix__c",
                  ]
                }
                mask="prefix__d"
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        numberOfSpaces={18}
        style={
          Array [
            Object {
              "height": 72,
            },
          ]
        }
      />
      <Text
        color="#ffffff"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Montserrat-ExtraBoldItalic",
              "fontSize": 28,
              "fontWeight": "800",
              "lineHeight": 34,
            },
          ]
        }
      >
        Bienvenue !
      </Text>
      <View
        numberOfSpaces={2}
        style={
          Array [
            Object {
              "height": 8,
            },
          ]
        }
      />
      <Text
        color="#ffffff"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 15,
              "fontWeight": "normal",
              "lineHeight": 20,
            },
          ]
        }
      >
        Toute la culture dans votre main
      </Text>
      <View
        numberOfSpaces={8}
        style={
          Array [
            Object {
              "height": 32,
            },
          ]
        }
      />
      <View
        numberOfSpaces={6}
        style={
          Array [
            Object {
              "height": 24,
            },
          ]
        }
      />
      <Text>
        Bienvenue à Pass Culture
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Aller sur la page de connexion avec params
          </Text>
        </View>
      </View>
      <View
        numberOfSpaces={6}
        style={
          Array [
            Object {
              "height": 24,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "paddingVertical": 20,
            },
          ]
        }
      >
        <Text>
          EMAIL: john.doe@passculture.app
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="refreshUserInformationsButton"
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Rafraîchir les données de l'utilisateur
            </Text>
          </View>
        </View>
      </View>
      <View
        numberOfSpaces={6}
        style={
          Array [
            Object {
              "height": 24,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
            },
            Array [
              Object {
                "alignItems": "center",
                "bottom": 50,
                "position": "absolute",
              },
            ],
          ]
        }
      >
        <Text
          style={
            Object {
              "fontSize": 9,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#0066ff",
              }
            }
          >
             Lat 
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 9,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#0066ff",
              }
            }
          >
             Lon 
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 9,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#0066ff",
              }
            }
          >
             Alt 
          </Text>
        </Text>
      </View>
      <View
        numberOfSpaces={6}
        style={
          Array [
            Object {
              "height": 24,
            },
          ]
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
